common/src/main/java/io/pravega/common/io/BoundedInputStream.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.io;
+
+import com.google.common.base.Preconditions;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import lombok.Getter;
+
+/**
+ * A wrapper for an InputStream that allows reading up to a specified number of bytes.
+ */
+public class BoundedInputStream extends FilterInputStream {
```

Comment:
This class was moved from `io.pravega.common.io.serialization.RevisionDataInputStream` into its own file.